mat-process-api/V1/Controllers/ProcessDataController.cs

```diff
@@ -54,21 +58,41 @@ public IActionResult GetProcessData(string processRef)
         /// If processData needs to be update, it's instead replaced with a full new one.
         /// Returning flat out 200 on successful update
         /// </summary>
-        /// <param name="processDataJSON"></param>
+        /// <param name="updateRequest"></param>
         /// <returns></returns>
         [HttpPatch]
+        [Route("{processRef}")]
         [Produces("application/json")]
         [ProducesResponseType(StatusCodes.Status200OK)]
-        public IActionResult UpdateExistingProcessDocument()
+        public IActionResult UpdateExistingProcessDocument([FromBody]UpdateProcessDataRequest updateRequest)
         {
-            return Ok();
+            var isRequestValid = _updateValidator.Validate(updateRequest);
+            if (isRequestValid.IsValid)
+            {
+
+                try
+                {
+                    _logger.LogInformation($"Update ProcessData request for process ID {updateRequest.processRef}");
+                    var result = _processDataUsecase.ExecuteUpdate(updateRequest);
+                    return Ok(result);
+                }
+                catch (DocumentNotFound ex)
+                {
+                    return StatusCode(200, $"Document with reference {updateRequest.processRef} was not found in the database." +
```

Comment:
The "Success error" question has been left for a discussion some other time. 
It was decided to leave this as is, as there's a lack of time discuss something as minor as this.